### PR TITLE
build(test-service-load): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution

Tests remain using CJS. [AB#7443](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7443) tracks conversion to ESM testing.

### CODE REVIEW

**Comment by RishhiB** [comment]
Is there a difference between scopes `@fluid-internal` and `@fluid-private`

File: packages/test/test-service-load/package.json [hunk 1]
```diff
@@ -59,12 +59,12 @@
 		"cache-dir": "nyc/.cache",
 		"exclude": [
 			"src/test/**/*.*ts",
-			"dist/test/**/*.*js"
+			"lib/test/**/*.*js"
```
Comment (jason-ha) [inline packages/test/test-service-load/package.json:62]: Lots of things left using dist in here. See package.json scripts, .mocharc.cjs, and src/test/mini.spec.ts that launches `./dist/nodeStressTest.js`.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: packages/test/test-service-load/package.json (hunk 1 above)
Comment (DLehenbauer) [reply packages/test/test-service-load/package.json]: Switched c8 back to /dist/ and created AB#7443 to track fully converting to ESM.

**Review by jason-ha — APPROVED** [review]

**Comment by tylerbutler** [comment]
> > Is there a difference between scopes `@fluid-internal` and `@fluid-private`
> 
> @RishhiB, I think the difference is that internal mean internal to collection of publicly published FF packages. private isn't or isn't necessarily published publicly. @tylerbutler / @ChumpChief, is that correct?

See [the eng.ms docs](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/dev/wiki/sync/npm-package-scopes) for details on the scopes.

**Comment by jason-ha** [comment]
> > > Is there a difference between scopes `@fluid-internal` and `@fluid-private`
> > 
> > 
> > @RishhiB, I think the difference is that internal mean internal to collection of publicly published FF packages. private isn't or isn't necessarily published publicly. @tylerbutler / @ChumpChief, is that correct?
> 
> See [the eng.ms docs](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/dev/wiki/sync/npm-package-scopes) for details on the scopes.

eng docs don't cover `@fluid-private` and I think it is wrong. It states `@fluid-internal` is not published, but at least some are like `@fluid-internal/client-utils`.

**Comment by alexvy86** [comment]
I'm reverting this as an immediate remediation to the fact that it broke stress tests (logs [here](https://dev.azure.com/fluidframework/internal/_build/results?buildId=246095&view=logs&j=e50769d2-5ed6-5a80-8b91-9bada4aff639&t=9ce5923c-5e55-5628-96fb-8b88f8e33486&l=24)) because [it does dynamic importing of a package by its folder, not all the way to a `.js` file](https://github.com/alexvy86/FluidFramework/blob/e1fab4e86ef75ea42b0306aa89c73594b50377d1/packages/test/test-service-load/src/utils.ts#L49). [Related PR](https://github.com/microsoft/FluidFramework/pull/20058) that fixes it for azure scenario runner. I didn't know we also did it here, but the same solution seems necessary, I'll update that PR. And would appreciate any further eyes on it :).

**Pull request merged**